### DISCUSSION TITLE
k3s v1.20.15 How to add etcd auto-compaction-retention  Isoparametric

### DISCUSSION THREAD

**Discussion by susugo** [k3s-io/k3s#8108] [Q&A] [answered]
@brosner See (https://github.com/k3s-io/k3s/issues/4787#issuecomment-997444292)When it comes to automatic compression, does it not need to add the auto-compaction deletion parameter? It is only available in that version, and how should it be verified;

**Comment by brandond** [comment]
I'm not quite sure what you're asking. As that comment says, Kubernetes automatically compacts the datastore without any extra configuration. The issue you linked to was related to exceeding the etcd quota.

**Reply by brandond** [reply]
I will also note that v1.20 is long, long end of life. You should upgrade immediately.

**Reply by susugo** [reply]
In ETCD, it is necessary to add the `--auto-compaction-retention` parameter for compression, as shown on the ETCD official website（https://etcd.io/docs/v3.5/op-guide/maintenance/#auto-compaction） 

I would like to know how Kubernetes is automatically implemented, as I see that the embedded ETCD does not have this parameter. How can I ensure that it will compress

**Reply by brandond** [reply, marked as answer]
You don't need to use that etcd option when using Kubernetes. The apiserver compacts the datastore for you every 5 minutes - reference the apiserver CLI flag:

https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/

> --etcd-compaction-interval duration     Default: 5m0s
>  The interval of compaction requests. If 0, the compaction request from apiserver is disabled.
